frontend/src/utils/ethereum.js

```diff
@@ -1,87 +1,133 @@
-import DecentradeMarketplaceAbi from '../../../smart-contracts/artifacts/contracts/Marketplace.sol/DecentradeMarketplace.json';
-import {
-    testIpfs,
-    uploadMetadataToIPFS,
-    uploadToIPFS,
-} from '../services/pinataService';
+const nftAddress = import.meta.env.VITE_NFT_ADDRESS
+const marketplaceAddress = import.meta.env.VITE_MARKET_ADDRESS
 
-const nftAddress = import.meta.env.VITE_NFT_ADDRESS;
-const marketplaceAddress = import.meta.env.VITE_MARKET_ADDRESS;
-
-console.log('marketplaceAddress:', marketplaceAddress);
-console.log('nftAddress:', nftAddress);
+console.log('marketplaceAddress:', marketplaceAddress)
+console.log('nftAddress:', nftAddress)
 
 export const connectWallet = async () => {
     if (window.ethereum) {
         try {
-            await window.ethereum.request({ method: 'eth_requestAccounts' });
-            const provider = new ethers.BrowserProvider(window.ethereum);
-            const signer = await provider.getSigner();
-            console.log('Wallet connected:', await signer.getAddress());
-            return signer;
+            await window.ethereum.request({ method: 'eth_requestAccounts' })
+            const provider = new ethers.BrowserProvider(window.ethereum)
+            const signer = await provider.getSigner()
+            console.log('Wallet connected:', await signer.getAddress())
+            return signer
         } catch (error) {
-            console.error('Error connecting wallet:', error);
-            throw error;
+            console.error('Error connecting wallet:', error)
+            throw error
         }
     } else {
-        console.error('Ethereum wallet not detected');
-        throw new Error('Ethereum wallet not detected');
+        console.error('Ethereum wallet not detected')
+        throw new Error('Ethereum wallet not detected')
     }
-};
+}
 
 export const getNFTContract = (signer) => {
-    return new ethers.Contract(nftAddress, DecentradeNFTAbi.abi, signer);
-};
+    try {
+        return new ethers.Contract(nftAddress, DecentradeNFTAbi.abi, signer)
+    } catch (e) {
+        console.error(e)
+    }
+}
 
 export const getMarketplaceContract = (signer) => {
-    return new ethers.Contract(marketplaceAddress, DecentradeMarketplaceAbi.abi, signer);
-};
-
+    return new ethers.Contract(
+        marketplaceAddress,
+        DecentradeMarketplaceAbi.abi,
+        signer
+    )
+}
+const uploadToIPFS = async (file) => {
+    try {
+        const formData = new FormData()
+        formData.append('file', file)
+        const response = await fetch('http://localhost:5000/ipfs/uploadImage', {
+            method: 'POST',
+            body: formData,
+        })
+        const data = await response.json()
+        console.log('IPFS Response:', data)
+        return data.ipfsHash
+    } catch (e) {
+        console.error(e)
+    }
+}
+const uploadMetadataToIPFS = async (metadata) => {
```

Comment:
Any code paths that do not have explicit returns will return undefined. It is recommended to replace any implicit dead-ends that return undefined with a return null statement.
```html
Bad Practice

function getUser(name) {
    if (userExists(name)) {
        return userDb.get(name);
    }
    // whoops! Implicit undefined return
}

function readFile(fileName) {
    if (fs.existsSync(fileName)) {
        return fs.readFileSync(fileName, 'utf-8');
    }

    return; // Implicit undefined return.
}

Recommended

function getUser(name) {
    if (userExists(name)) {
        return userDb.get(name);
    }

    // `null` indicates there is no such user.
    return null;
}

function readFile(fileName) {
    if (fs.existsSync(fileName)) {
        return fs.readFileSync(fileName, 'utf-8');
    }

    // `null` indicates there is no such file.
    return null;
}



```